Automattic/wp-calypso
# Remove the Redux navigation middleware and update Activity Log filter actions

Removes the `state/navigation/middleware` Redux middleware. It was intercepting the `ACTIVITY_LOG_FILTER_{SET,UPDATE}` actions and updated the URL query string according to the latest value of the activity log filter.

This PR moves that navigation to the actions themselves, by making them into thunks.

**How to test:**
Open Activity Log for a site that has it active (requires a paid plan) and set a filter on the log:

<img width="710" alt="Screenshot 2021-06-08 at 9 21 37" src="https://user-images.githubusercontent.com/664258/121141583-6d75e280-c83b-11eb-9e66-aa15d4fd07d4.png">

Verify that the filter is really activated (actually filtering the log) and that also the URL is updated with something like:
```
?group=post
```

Note that the current implementation is a bit suboptimal: on filter change, it updates the Redux state, rerendering the UI, and also triggers a _page.js_ navigation with an updated URL which triggers a second, redundant rerender. That could be optimized by either:
- not updating the Redux state on filter change, but merely navigating to a new URL. The route handlers activate the new filter.
- updating the Redux state and doing `page.replace` without dispatch, updating the URL without triggering any navigation consequences.

But I intentionally avoided touching that.
